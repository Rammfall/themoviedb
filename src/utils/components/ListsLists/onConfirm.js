/**
 * @param {number} id
 * @param {function} callback
 * @returns {function(): *}
 */
const onConfirm = (id, callback) => () => {
  callback({ id })
}

export default onConfirm
