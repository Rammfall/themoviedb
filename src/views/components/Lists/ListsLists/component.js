import React from 'react'
import { Col, Row } from 'antd'
import PropTypes from 'prop-types'

import ListCard from 'Views/components/Lists/ListsLists/ListCard'
import ListsPagination from './ListsPagination'

const ListsLists = ({ lists }) => (
  <>
    <Row
      gutter={8}
      type='flex'
    >
      <Col
        span={20}
        offset={2}
      >
        <Row
          gutter={{
            xs: 8,
            sm: 16,
            md: 24,
            lg: 32
          }}
        >
          {lists.map(({name, description, id}) => (
            <ListCard
              key={id}
              name={name}
              description={description}
            />)
          )}
        </Row>
      </Col>
    </Row>
    <Row
      type='flex'
      justify='center'
    >
      <Col>
        <ListsPagination />
      </Col>
    </Row>
  </>
)

ListsLists.propTypes = {
  lists: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    description: PropTypes.string
  }).isRequired).isRequired
}

export default ListsLists
