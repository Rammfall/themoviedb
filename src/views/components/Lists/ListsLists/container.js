import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Modal } from 'antd'
import { injectIntl } from 'react-intl'
import { compose } from 'ramda'
import { withRouter } from 'react-router-dom'

import onConfirm from 'Utils/components/modalsHandlers/onConfirm'
import { getListsSelector, getListsTotalSelector } from 'Store/concepts/lists/selectors'
import { deleteList } from 'Store/concepts/lists/actions'
import { listsConstant } from 'Constants/concepts'

import ListsListsComponent from './component'

class ListsLists extends Component {
  onDeleteModal = (id) => {
    const { onDelete, intl: { formatMessage } } = this.props

    Modal.confirm({
      title: formatMessage({ id: 'lists.delete.title' }),
      onOk: onConfirm(id, onDelete)
    })
  }

  onCardClick = (id) => () => {
    const { history: { push } } = this.props

    push({
      pathname: `${listsConstant}/${id}`
    })
  }

  render() {
    const { total, lists } = this.props

    return (
      <ListsListsComponent
        total={total}
        lists={lists}
        onDelete={this.onDeleteModal}
        onClick={this.onCardClick}
      />
    )
  }
}

ListsLists.propTypes = {
  lists: PropTypes.arrayOf(PropTypes.shape({
    description: PropTypes.string,
    name: PropTypes.string,
    id: PropTypes.number
  })).isRequired,
  total: PropTypes.number,
  onDelete: PropTypes.func.isRequired,
  intl: PropTypes.shape({
    formatMessage: PropTypes.func.isRequired
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
}

ListsLists.defaultProps = {
  total: null
}

const mapStateToProps = (state) => ({
  total: getListsTotalSelector(state),
  lists: getListsSelector(state)
})

const mapDispatchToProps = {
  onDelete: deleteList
}

export { ListsLists as ListsListsContainer }
export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  injectIntl,
  withRouter
)(ListsLists)
