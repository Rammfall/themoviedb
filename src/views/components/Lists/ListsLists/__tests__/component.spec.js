import React from 'react'
import { shallow } from 'enzyme'

import ListsLists from '../component'

describe('ListsLists', () => {
  it('matches snapshot', () => {
    const props = {
      lists: [
        {
          id: 1,
          name: 'test',
          description: 'test'
        }
      ]
    }
    const component = shallow(<ListsLists {...props} />)

    expect(component).toMatchSnapshot()
  })
})
