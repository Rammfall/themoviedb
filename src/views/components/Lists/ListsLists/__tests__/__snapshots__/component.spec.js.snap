// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListsLists matches snapshot 1`] = `
<Fragment>
  <Row
    gutter={8}
    type="flex"
  >
    <Col
      offset={2}
      span={20}
    >
      <Row
        gutter={
          Object {
            "lg": 32,
            "md": 24,
            "sm": 16,
            "xs": 8,
          }
        }
      >
        <ListCard
          description="test"
          key="1"
          name="test"
        />
      </Row>
    </Col>
  </Row>
  <Row
    justify="center"
    type="flex"
  >
    <Col>
      <Connect(ListsPagination) />
    </Col>
  </Row>
</Fragment>
`;
